### DISCUSSION TITLE
How do I get a client used for ethers? how do I sign messages?

### DISCUSSION THREAD

**Discussion by liron-navon** [WalletConnect/walletconnect-monorepo#3029] [Developer Support] [answered]
I'm unable to see how I can get the client used by web3modal, I want to use ethers to sign messages and transactions but the modal doesn't expose a provider for use?

**Comment by crypblizz8** [comment, marked as answer]
Hey there if you are using the web3modal, we recommend using wagmi as that is all inclusive.

If you are using ethereum/universal provider, it is still possible. Our example with ethers.js is here.
https://github.com/WalletConnect/web-examples/tree/main/dapps/react-dapp-v2-with-ethers
